test/src/test/java/hudson/model/RSSTest.java

```diff
@@ -63,16 +64,89 @@ public void absoluteURLsPresentInRSS_evenWithoutRootUrlSetup() throws Exception
+        Node channelNode = documentElement.getFirstChild();
+        assertThat(channelNode.getNodeName(), is("channel"));
+        checkRssBasicNodes(channelNode, "All failed builds", 3);
+    }
+
+    @Test
+    public void checkInitialContentAllAtom() throws Exception {
+        XmlPage page = getRssAllAtomPage();
+        Document xmlDocument = page.getXmlDocument();
+        Element documentElement = xmlDocument.getDocumentElement();
+        assertThat(documentElement.getNodeName(), is("feed"));
+        checkAtomBasicNodes(documentElement, "All builds", 5);
+    }
+
+    @Test
+    public void checkWithSingleBuildAllRss() throws Exception {
+        FreeStyleProject p = j.createFreeStyleProject();
+        j.assertBuildStatusSuccess(p.scheduleBuild2(0));
+
+        XmlPage page = getRssAllPage();
+        Document xmlDocument = page.getXmlDocument();
+        Element documentElement = xmlDocument.getDocumentElement();
+        assertThat(documentElement.getNodeName(), is("rss"));
+        assertThat(documentElement.getAttribute("version"), is("2.0"));
+        assertThat(documentElement.getChildNodes().getLength(), is(1));
+        Node channelNode = documentElement.getFirstChild();
+        assertThat(channelNode.getNodeName(), is("channel"));
+        checkRssBasicNodes(channelNode, "All builds", 4);
+        NodeList items = xmlDocument.getElementsByTagName("item");
+        assertThat(items.getLength(), is(1));
+        Node firstBuild = items.item(0);
+        assertThat(firstBuild.getChildNodes().getLength(), is(5));
+        assertThat(getSingleNode(firstBuild, "title").getTextContent(), is("test0 #1 (stable)"));
+        assertNotNull(getSingleNode(firstBuild, "pubDate").getTextContent());
+        assertNotNull(getSingleNode(firstBuild, "author").getTextContent());
+        Node guidNode = getSingleNode(firstBuild, "guid");
+        assertThat(guidNode.getAttributes().getNamedItem("isPermaLink").getTextContent(), is("false"));
+        assertNotNull(guidNode.getTextContent());
+    }
+
+    @Test
+    public void checkWithSingleBuildAllAtom() throws Exception {
+        FreeStyleProject p = j.createFreeStyleProject();
+        j.assertBuildStatusSuccess(p.scheduleBuild2(0));
+
+        XmlPage page = getRssAllAtomPage();
+        Document xmlDocument = page.getXmlDocument();
+        Element documentElement = xmlDocument.getDocumentElement();
+        assertThat(documentElement.getNodeName(), is("feed"));
+        checkAtomBasicNodes(documentElement, "All builds", 6);
+        NodeList entries = xmlDocument.getElementsByTagName("entry");
+        assertThat(entries.getLength(), is(1));
+        Node firstBuild = entries.item(0);
+        assertThat(firstBuild.getChildNodes().getLength(), is(5));
+        assertThat(getSingleNode(firstBuild, "title").getTextContent(), is("test0 #1 (stable)"));
+        assertNotNull(getSingleNode(firstBuild, "published").getTextContent());
```

Comment:
:x: Does not cover the regression corrected by https://github.com/jenkinsci/jenkins/pull/4148. The "published" was empty and so, is still not null.

Reply:
Added checks to validate the dates. There are advantages and disadvantages to canned data. It can mask or complicate other things. There isn't any real need for canned data here.